test/conic_tests.jl

```diff
@@ -0,0 +1,213 @@
+
+    cached = MOIU.CachingOptimizer(MOIU.UniversalFallback(MOIU.Model{Float64}()), ECOS.Optimizer())
+    optimizer = POI.ParametricOptimizer(cached)
+
+    model = direct_model(optimizer);
+    @variable(model, x)
+    @variable(model, y)
+    @variable(model, t in POI.Parameter(5))
+    @constraint(model, [(x - t + 1), (y - t + 2)...] in MOI.Nonnegatives(2));
+    @objective(model, Min, x + y)
+    optimize!(model)
+
+    @test isapprox.(value(x), 4.0, atol = ATOL)
+    @test isapprox.(value(y), 3.0, atol = ATOL)
+
+    MOI.set(model, POI.ParameterValue(), t, 6)
+    optimize!(model)
+
+    @test isapprox.(value(x), 5.0, atol = ATOL)
+    @test isapprox.(value(y), 4.0, atol = ATOL)
+end
+
+@testset "CONIC - SOC - Parameter in affine part" begin
+    """
+        Problem SOC2 from MOI
+
+        min  x
+        s.t. y ≥ 1/√2
+            (x-p)² + y² ≤ 1
+
+        in conic form:
+
+        min  x
+        s.t.  -1/√2 + y ∈ R₊
+            1 - t ∈ {0}
+            (t, x-p ,y) ∈ SOC₃
+
+        opt 
+            x* = p - 1/√2
+            y* = 1/√2
+    """
+    cached = MOIU.CachingOptimizer(MOIU.UniversalFallback(MOIU.Model{Float64}()), ECOS.Optimizer())
+    model = POI.ParametricOptimizer(cached)
+    
+    x, y, t = MOI.add_variables(model, 3)
+    p, cp = MOI.add_constrained_variable(model, POI.Parameter(0))
+    
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+        MOI.ScalarAffineFunction([MOI.ScalarAffineTerm(1.0, x)], 0.0),
+    )
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    
+    cnon = MOI.add_constraint(
+        model,
+        MOI.VectorAffineFunction(
+            [MOI.VectorAffineTerm(1, MOI.ScalarAffineTerm(1.0, y))],
+            [-1 / √2],
+        ),
+        MOI.Nonnegatives(1),
+    )
+    
+    ceq = MOI.add_constraint(
+        model,
+        MOI.VectorAffineFunction(
+            [MOI.VectorAffineTerm(1, MOI.ScalarAffineTerm(-1.0, t))],
+            [1.0],
+        ),
+        MOI.Zeros(1),
+    )
+    
+    A = [1. 0. 0. 0.;
+         0. 1. 0. -1;
+         0. 0. 1. 0.]
+    f = MOI.VectorAffineFunction(
+        vec(MOI.VectorAffineTerm.(1:3, MOI.ScalarAffineTerm.(A, reshape([t, x, y, p], 1, 4)))),
+        zeros(3)
+    )
+    csoc = MOI.add_constraint(
+        model,
+        f,
+        MOI.SecondOrderCone(3),
+    )
+    
+    MOI.optimize!(model)
+    
+    @test MOI.get(model, MOI.ObjectiveValue()) ≈ -1 / √2 atol = ATOL
+    @test MOI.get(model, MOI.VariablePrimal(), x) ≈ -1 / √2 atol = ATOL
+    @test MOI.get(model, MOI.VariablePrimal(), y) ≈ 1 / √2 atol = ATOL
+    @test MOI.get(model, MOI.VariablePrimal(), t) ≈ 1 atol = ATOL
+    
+    MOI.set(model, POI.ParameterValue(), p, 1)
+    MOI.optimize!(model)
+    
+    @test MOI.get(model, MOI.ObjectiveValue()) ≈ 1 - 1/√2 atol = ATOL
+    @test MOI.get(model, MOI.VariablePrimal(), x) ≈ 1 - 1/√2 atol = ATOL
+end
+
+# @testset "JuMP direct model - CONIC - SOC - Parameter in affine part" begin
```

Comment:
uncomment?

Reply:
I can't get to work the constraint `t-1 in Zeros(1)`, that's why I commented it

Do you know how to properly do it?

Reply:
maybe `[t-1] in Zeros(1)`?